src/main/java/org/mitre/synthea/engine/State.java

```diff
@@ -1300,12 +1300,45 @@ public MedicationOrder clone() {
 
     @Override
     public boolean process(Person person, long time) {
+      Medication medication = null;
+      boolean createPrescription = true;
       String primaryCode = codes.get(0).code;
-      Medication medication = person.record.medicationStart(time, primaryCode, chronic);
+      if (this.administration) {
+        medication = person.record.medicationAdministration(time, primaryCode, codes);
+        applyFeatures(person, medication);
+
+        if (!this.chronic) {
+          createPrescription = false;
+        }
+      }
+      if (createPrescription) {
+        medication = person.record.medicationStart(time, primaryCode, chronic);
+        applyFeatures(person, medication);
+      }
+      // increment number of prescriptions prescribed by respective hospital
+      Provider medicationProvider = person.getCurrentProvider(module.name);
+      if (medicationProvider == null) {
+        // no provider associated with encounter or medication order
+        medicationProvider = person.getProvider(EncounterType.WELLNESS, time);
+      }
+
+      int year = Utilities.getYear(time);
+      medicationProvider.incrementPrescriptions(year);
+      return true;
+    }
+
+    /**
+     * Apply all the various features to the medication.
+     * @param person the Person.
+     * @param medication the Medication.
+     */
+    private void applyFeatures(Person person, Medication medication) {
```

Comment:
Could this get code get moved into `HealthRecord` and then possibly expand the signature of `medicationStart` and `medicationAdministration`? If a medication is an administration and chronic, `applyFeatures` gets called twice on the same object. It's harmless, but it feels like there is a better way to unwind all of this.

Reply:
I could move this if you really want, but it is NOT calling `applyFeatures` twice on the same object.

Instead, it is creating one or two medication objects in the HealthRecord, and calling applyFeatures on each one. I put the logic here, because the logic potentially looks at prior state history that seems outside the scope of HealthRecord.

If the state has the `administration` flag, the Medication is an administration. (In FHIR this one becomes a `MedicationAdministration` resource).

If the state also results in a prescription (the default behavior, but also possible if it is a _chronic_ administration), another Medication is added into the HealthRecord (if the prescription does not already exist). (In FHIR this one becomes a `MedicationRequest` resource).

So, a Medication state can result in three outcomes:

- MedicationRequest
- MedicationAdministration
- MedicationRequest && MedicationAdministration